### PR TITLE
US136813 Dynamize Localizations

### PR DESCRIPTION
**Pull request by ChrisLabattD2L**
Context for [US136813](https://rally1.rallydev.com/#/?detail=/userstory/628440009685&fdp=true): Simplifying and Dynamizing Localizations

Since this repo already uses the dynamic mixin, all we have to do is replace the [old localization mixin link](https://github.com/BrightspaceUI/core/blob/main/mixins/localize-dynamic-mixin.js) with the [new one](https://github.com/BrightspaceUI/core/blob/main/mixins/localize/localize-mixin.js). The old one simply just points to the new one, but will likely get removed at some point, so we're just updating things ahead of time.

I also renamed the `localize-mixin` file to `localization`, to keep it consistent with our other labs repos.

Note: due to an [update with Playwright](https://d2l.slack.com/archives/C0PHG3QB0/p1694663113104319), I also had to update the CI workflow.

### CODE REVIEW

File: localization.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { LocalizeMixin } from '@brightspace-ui/core/mixins/localize/localize-mixin.js';
+
+export const Localizer = superClass =>
```
Comment (tenbradley) [inline localization.js:3]: The naming here is the only part of this pr I am concerned about. This is a fine name. I've just never seen another repo call it something like this and I wonder if a name that is more consistent with other repos would be good.

**Review by tenbradley — APPROVED** [review]
The only comment I have is more a nit than a real concern. I have no problem with it being merged as is if you want to keep this name.

File: localization.js (hunk 1 above)
Comment (ChrisLabattD2L) [reply localization.js:3]: This is mostly just what it's called in [multi-select](https://github.com/BrightspaceUILabs/multi-select/blob/main/localization.js) and [pagination](https://github.com/BrightspaceUILabs/pagination/blob/main/lang/localization.js) labs repos, so I figured I've make it consistent with them.

Comment (tenbradley) [reply localization.js:3]: This is acceptable.

Comment (ChrisLabattD2L) [reply localization.js:3]: I am most gracious of this acceptance.

**Pull request merged**